server/__tests__/signupAPI.test.js

```diff
@@ -0,0 +1,96 @@
+import { expect } from 'chai';
+import supertest from 'supertest';
+import mongoose from 'mongoose';
+import { Mockgoose } from 'mockgoose';
+import app from '../src/app/app';
+import getMockUser from '../src/utils/testUtils';
+import { TEST_PORT, DB_URL } from '../src/app/appConstants';
+
+let mockUser;
+let testServer;
+let request;
+let mockgoose;
+
+beforeAll(() => {
+  mockgoose = new Mockgoose(mongoose);
+  mockgoose.prepareStorage().then(() => {
+    mongoose.connect(DB_URL, { useNewUrlParser: true, useCreateIndex: true });
+  });
+});
+
+beforeEach((done) => {
+  mockUser = getMockUser();
+  testServer = app.listen(TEST_PORT, () => {
+    request = supertest(testServer);
+    return done();
+  });
+});
+
+afterEach(done => testServer.close(done));
+
+afterAll(() => mockgoose.shutdown());
+
+// /api/signup test suite
+describe('Sign-Up API', () => {
+  it('registers a user with email, username and password', (done) => {
+    request
+      .post('/api/signup')
+      .send(mockUser)
+      .expect(200)
+      .end((err, res) => {
+        const createdUser = res.body;
+        expect(createdUser).to.have.keys('id', 'username');
+        return done(err);
+      });
+  });
+  it('give error codes when email or username or password are missing', (done) => {
+    const { email, username } = mockUser;
+    request
+      .post('/api/signup')
+      .send({
+        email,
+        username,
+      })
+      .expect(422)
+      .end(() => done());
+  });
+  it('gives error messages when email or username or password are missing', (done) => {
```

Comment:
I think you can check status code and error message in one test.

Reply:
@niinpatel @nimish-gupta Thoughts?